ext/semian/sysv_semaphores.c

```diff
@@ -80,19 +81,27 @@ set_semaphore_permissions(int sem_id, long permissions)
 }
 
 int
-perform_semop(int sem_id, short index, short op, short flags, struct timespec *ts)
+perform_semop(int sem_id, short index, short op, short flags, struct timespec *ts, int *wait_time_ms)
```

Comment:
I think it could make sense to just define a type which includes both timespec and an additional int for wait_time_ms, though either way the call signature would have to change.

I think having a single argument for timing related info is cleaner, but I also thing the separate variable might be more apparent. 

something to watch though, as this function alright takes a lot of parameters.